Added reboot check before attempting to provision

sneal opened the pull request:
If the guest is shutting down don't start the chef provisioning run, wait for the machine to come back up in a ready to communicate state.

This solves the problem where chef provisioning may try to start while the machine is shutting down. Depending on the timing this can lead to the dna.json file not being found or at the very least requiring chef to try another attempt.

As a real world example this fixes our CI build of our VisualStudio cookbook that has 2 provisioning steps
1. Install .NET 4.5 & reboot
2. Install Visual Studio

sneal committed:
Added reboot check before attempting to provision

kashook commented:
@sneal, I built a copy of this and have been testing it out.  In my post [here](https://github.com/WinRb/vagrant-windows/pull/121) I described some of the stuff that's causing me to reboot.  I realized later that several of our cookbooks (such as SQL server) detect whether the installation returns an exit status that indicates that a reboot is needed, and if so, then they use the reboot resource in the [windows cookbook](http://community.opscode.com/cookbooks/windows)  to reboot the system.  The actual reboot happens with a `shutdown /r /t` command, which is issued at the end of the chef run.

I broke apart the chef provisioner in my Vagrantfile into several different chef provisioners such that the items that could potentially want a reboot (such as SQL and .NET) would be in their own provisioner.  That way, the system would reboot at the end of the chef run if needed, and I wouldn't have to do any powershell stuff to "manually" reboot in between the chef runs.  Sure enough, SQL server wants a reboot, and a `shutdown` command did get issued with a timeout of 60 seconds.  Here are the results.  (I had info logging enabled, so the chef log messages are doubled).

This log snippet begins at the very end of the chef run that installed SQL server and issued a shutdown command.

```
 ...

 INFO interface: info: [2013-10-28T12:36:29-04:00] INFO: Chef Run complete in 1294.052734 seconds
[2013-10-28T12:36:29-04:00] INFO: Chef Run complete in 1294.052734 seconds INFO interface: info:


 INFO interface: info: [2013-10-28T12:36:29-04:00] INFO: Running report handlers
[2013-10-28T12:36:29-04:00] INFO: Running report handlers INFO interface: info:
[2013-10-28T12:36:29-04:00] WARN: chef_handler[WindowsRebootHandler] requested reboot will occur in 60 seconds


[2013-10-28T12:36:29-04:00] WARN: chef_handler[WindowsRebootHandler] requested reboot will occur in 60 seconds
 INFO interface: info: [2013-10-28T12:36:29-04:00] INFO: Report handlers complete
[2013-10-28T12:36:29-04:00] INFO: Report handlers complete INFO interface: info:
```

The next log snippet shows that the next chef run started.  The system had not yet rebooted (confirmed by the timestamps on the log messages).  It does reboot after the chef run has kicked off, and a WinRMHTTPTransportError is produced.

```
 INFO interface: info: Running provisioner: chef_solo...
[cswin] Running provisioner: chef_solo...
 INFO chef_solo: Checking guest reboot status
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "showvminfo", "bbf02b54-cf12-406f-ba4a-5b92e4950bbd", "--machinereadable"]
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "showvminfo", "bbf02b54-cf12-406f-ba4a-5b92e4950bbd", "--machinereadable"]
 INFO interface: info: Generating chef JSON and uploading...
Generating chef JSON and uploading...
 INFO interface: info: Running chef-solo...
Running chef-solo...
 INFO interface: info: [2013-10-28T12:36:45-04:00] INFO: *** Chef 11.4.4 ***
[2013-10-28T12:36:45-04:00] INFO: *** Chef 11.4.4 *** INFO interface: info:


 INFO retryable: Retryable exception raised: #<WinRM::WinRMHTTPTransportError: Bad HTTP response returned from server (401).>
```

After this error, the same chef run was kicked off again.  This time, it errors off with the "CookbookNotFound" error.

```
 INFO interface: info: [2013-10-28T12:37:59-04:00] INFO: *** Chef 11.4.4 ***
[2013-10-28T12:37:59-04:00] INFO: *** Chef 11.4.4 *** INFO interface: info:


 INFO interface: info: [2013-10-28T12:38:55-04:00] INFO: Setting the run_list to [OBFUSCATED] from JSON
[2013-10-28T12:38:55-04:00] INFO: Setting the run_list to [OBFUSCATED] from JSON INFO interface: info:


 INFO interface: info: [2013-10-28T12:38:55-04:00] INFO: Run List is [OBFUSCATED]
[2013-10-28T12:38:55-04:00] INFO: Run List is [OBFUSCATED] INFO interface: info:
[2013-10-28T12:38:55-04:00] INFO: Run List expands to [OBFUSCATED]


[2013-10-28T12:38:55-04:00] INFO: Run List expands to [OBFUSCATED]
 INFO interface: info: [2013-10-28T12:38:55-04:00] INFO: Starting Chef Run for [OBFUSCATED]
[2013-10-28T12:38:55-04:00] INFO: Starting Chef Run for [OBFUSCATED] INFO interface: info:
[2013-10-28T12:38:55-04:00] INFO: Running start handlers


[2013-10-28T12:38:55-04:00] INFO: Running start handlers
 INFO interface: info: [2013-10-28T12:38:55-04:00] INFO: Start handlers complete.
[2013-10-28T12:38:55-04:00] INFO: Start handlers complete. INFO interface: info:


 INFO interface: info: [2013-10-28T12:38:55-04:00] ERROR: Running exception handlers
[2013-10-28T12:38:55-04:00] ERROR: Running exception handlers INFO interface: info:
[2013-10-28T12:38:55-04:00] ERROR: Exception handlers complete


[2013-10-28T12:38:55-04:00] ERROR: Exception handlers complete
 INFO interface: info: [2013-10-28T12:38:55-04:00] FATAL: Stacktrace dumped to C:/var/chef/cache/chef-stacktrace.out
[2013-10-28T12:38:55-04:00] FATAL: Stacktrace dumped to C:/var/chef/cache/chef-stacktrace.out INFO interface: info:


 INFO interface: info: [2013-10-28T12:38:55-04:00] FATAL: Chef::Exceptions::CookbookNotFound: Cookbook [OBFUSCATED] not found. If you're loading [OBFUSCATED] from another cookbook, make sure you configure the dependency in your metadata
[2013-10-28T12:38:55-04:00] FATAL: Chef::Exceptions::CookbookNotFound: Cookbook [OBFUSCATED] not found. If you're loading [OBFUSCATED] from another cookbook, make sure
you configure the dependency in your metadata INFO interface: info:


ERROR warden: Error occurred: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x74b9128>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Recovery complete.
ERROR warden: Error occurred: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x73c2b28>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Recovery complete.
ERROR warden: Error occurred: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x5fd5940>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x5fd59d0>
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<VagrantPlugins::ProviderVirtualBox::Action::Import:0x72cee98>
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "showvminfo", "bbf02b54-cf12-406f-ba4a-5b92e4950bbd", "--machinereadable"]
 INFO warden: Recovery complete.
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x5fd5bb0>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO environment: Running hook: environment_unload
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 1 hooks defined.
 INFO runner: Running action: #<Vagrant::Action::Builder:0x73d1540>
ERROR vagrant: Vagrant experienced an error! Details:
ERROR vagrant: #<VagrantPlugins::Chef::Provisioner::Base::ChefError: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.>
ERROR vagrant: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
ERROR vagrant: C:/Users/keiths/.vagrant.d/gems/gems/vagrant-windows-1.3.0.pre.1/lib/vagrant-windows/monkey_patches/plugins/provisioners/chef/provisioner/chef_solo.rb:80:in `run_chef_solo_on_windows'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-windows-1.3.0.pre.1/lib/vagrant-windows/monkey_patches/plugins/provisioners/chef/provisioner/chef_solo.rb:17:in `block in <class:ChefSolo>'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/provisioners/chef/provisioner/chef_solo.rb:54:in `provision'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-windows-1.3.0.pre.1/lib/vagrant-windows/monkey_patches/plugins/provisioners/chef/provisioner/chef_solo.rb:22:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-windows-1.3.0.pre.1/lib/vagrant-windows/monkey_patches/plugins/provisioners/chef/provisioner/chef_solo.rb:22:in `block in <class:ChefSolo>'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/provision.rb:74:in `run_provisioner'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/provision.rb:62:in `block in call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/provision.rb:58:in `each'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/provision.rb:58:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/providers/virtualbox/action/clear_forwarded_ports.rb:13:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/providers/virtualbox/action/set_name.rb:47:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/providers/virtualbox/action/clean_machine_folder.rb:17:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `block in run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/util/busy.rb:19:in `busy'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/call.rb:51:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `block in run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/util/busy.rb:19:in `busy'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/call.rb:51:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `block in run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/util/busy.rb:19:in `busy'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/call.rb:51:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/call.rb:57:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-chefutil-1.5.0-x86-mingw32/lib/vagrant-chefutil/action/copy_data_bags.rb:29:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/env_set.rb:19:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-chefutil-1.5.0-x86-mingw32/lib/vagrant-chefutil/action/copy_roles.rb:29:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/env_set.rb:19:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-berkshelf-1.3.5.ontsys.1/lib/berkshelf/vagrant/action/configure_chef.rb:23:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-berkshelf-1.3.5.ontsys.1/lib/berkshelf/vagrant/action/load_shelf.rb:28:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/vagrant-berkshelf-1.3.5.ontsys.1/lib/berkshelf/vagrant/action/set_ui.rb:12:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/env_set.rb:19:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builtin/call.rb:57:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builder.rb:116:in `call'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `block in run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/util/busy.rb:19:in `busy'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in `run'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/machine.rb:147:in `action'
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/batch_action.rb:63:in `block (2 levels) in run'
C:/Users/keiths/.vagrant.d/gems/gems/logging-1.8.1/lib/logging/diagnostic_context.rb:323:in `call'
C:/Users/keiths/.vagrant.d/gems/gems/logging-1.8.1/lib/logging/diagnostic_context.rb:323:in `block in create_with_logging_context'
C:/HashiCorp/Vagrant/embedded/lib/ruby/1.9.1/win32ole.rb:13:in `call'
C:/HashiCorp/Vagrant/embedded/lib/ruby/1.9.1/win32ole.rb:13:in `block in initialize'
 INFO interface: error: Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
```

After this approach didn't work, I tried using powershell the way I described in my [other post](https://github.com/WinRb/vagrant-windows/pull/121).  However, when I issued the `Restart-Computer` command, it gave me an error because the chef run had already issued a `shutdown /r` command.  I thought about reducing the timeout from 60 seconds to 0 (but it seems like I shouldn't depend on timing).  I also considered making the cookbooks not use windows_reboot if I set a certain option in the json so I could just use powershell outside of the chef provisioners.  However, I was hoping to get your thoughts on whether the approach I tried should have actually worked before making those changes.

sneal committed:
Fixed issue 119 on subsequent chef runs

need to reset the symbolic links when the box reboots during the middle of a chef run

sneal commented:
I was initially surprised to see the CookbookNotFound error after you rebooted, but then realized we need to reset the symbolic links before every chef attempt. I've fixed that and pushed a new commit for this branch.

Other than that I'm not surprised by the behavior you're seeing WRT the delayed reboot. I think we need to further modify the vagrant-windows shutdown detection code to somehow look for a pending reboot. Ideally we would be able to query Windows (registry somewhere?) for a pending reboot state from the shutdown command. If Windows doesn't then maybe the Chef Windows reboot handler stores state about a pending reboot?

For now I've hacked around the reboot problem by removing the reboot handler from my cookbooks and forcing a reboot via the shell provisioner. In general I'm not sure how to handle the case of running two cookbooks where a reboot must occur between them. I really wish I could have a cookbook force a reboot and Chef would automatically restart after the reboot (with or without Vagrant). Doing that correctly would be a rather large change to the chef client architecture, albeit not necessarily a bad one.

sneal committed:
Detect pending reboots

reboot_detect.ps1 now checks to see if Windows has a pending reboot scheduled to happen. This allows a provisioner to schedule a reboot only if needed using its builtin mechanisms.

ferventcoder commented on lib/vagrant-windows/scripts/reboot_detect.ps1:
```diff
@@ -0,0 +1,43 @@
+  
+  # Remove the pending reboot we just created above
+  if ($LASTEXITCODE -eq 0) {
+    . shutdown.exe -a
```
I always love how you have to try a reboot to see if a reboot is in progress.

ferventcoder commented:
Curious that this reboot detection is only with the chef-solo provisioner and not with the vagrant shell provisioner or other provisioners...is this just testing it out right now to ensure it works prior to opening up support for other provisioners?

sneal commented:
Yep, just testing it out for now.

kashook commented:
@sneal I've been testing out the updated reboot detection logic, and so far it seems to be working.  I went back to using the windows_reboot resource from the windows cookbook in my cookbooks and removed all the shell provisioner reboots.  Any recipe that might want a reboot was placed in its own chef provisioner.  I set a timeout of 60 seconds on the windows_reboot resource to make sure that vagrant-windows would get into the reboot detection logic.

sneal commented:
:+1: thanks for trying it out

sneal committed:
Bump version

sneal commented:
published this branch as 1.3.0.pre.2, can install via:

```
vagrant plugin install vagrant-windows --plugin-version 1.3.0.pre.2 --plugin-prerelease --plugin-source https://rubygems.org
```
